### PR TITLE
Feature/us 55 nfr refactor details

### PR DESCRIPTION
**Pull request by ATdllerenas**
-Refactor Details
-Implement Validation Get All Histories
-Junits Modifications 
-Resolve Errors

### CODE REVIEW

**Review by luisgalindoluna — APPROVED** [review]
Looks good

File: pom.xml [hunk 1]
```diff
@@ -44,7 +44,6 @@
 		<dependency>
 			<groupId>junit</groupId>
 			<artifactId>junit</artifactId>
-			<version>4.12</version>
```
Comment (ivanflores15) [inline pom.xml]: What version are we using?

Comment (luisgalindoluna) [reply pom.xml]: It's the same version that develop has right now

Comment (luisgalindoluna) [reply pom.xml]: No wait, nevermind. I misread the change.

Comment (ATdllerenas) [reply pom.xml]: I already added it again

**Review by ivanflores15 — APPROVED** [review]

**Pull request merged**